pkg/overlay/cache_test.go

```diff
@@ -80,8 +80,10 @@ func bootstrapTestNetwork(t *testing.T, ip, port string) ([]dht.DHT, pb.Node) {
 	assert.NoError(t, err)
 	bootNode := rt.Local()
 
-	err = boot.ListenAndServe()
-	assert.NoError(t, err)
+	go func() {
```

Comment:
Btw I created a `testcontext` package that helps with these sorts of tests.

```
ctx := testcontext.New(t)
defer ctx.Shutdown()
...
ctx.Go(boot.ListenAndServe)
```

Reply:
I'm not sure why, but using textcontext causes my tests to timeout.  Maybe a deadlock w/ the waitgroup?

ctx := testcontext.New(t)
defer ctx.Cleanup()
ctx.Go(boot.ListenAndServe)
^^ this fails

Reply:
This is would mean that you never shutdown the server.